GeneralsMD/Code/GameEngine/Source/Common/DamageFX.cpp

```diff
@@ -272,7 +272,7 @@ DamageFXStore::~DamageFXStore()
 }
 
 //-------------------------------------------------------------------------------------------------
-const DamageFX *DamageFXStore::findDamageFX(AsciiString name) const
+const DamageFX *DamageFXStore::findDamageFX(const char* name) const
 {
 	NameKeyType namekey = TheNameKeyGenerator->nameToKey(name);
   DamageFXMap::const_iterator it = m_dfxmap.find(namekey);
```

Comment:
I think this needs another look. If `m_dfxmap` has `AsciiString` key, then this allocates.

We already have a mechanism to solve this which we used in FileSystem for the file exist cache map.